### DISCUSSION TITLE
Save uploaded files in draft entries

### DISCUSSION THREAD

**Discussion by gmanu93** [harness-software/wp-graphql-gravity-forms#303] [Q&A] [answered]
I'm building a website in Next js that uses the save for later function in an extended formulary. When I submit an entry, it saves the uploaded files correctly. But when the entry is saved as a draft, the array for 'fileUploadValues' always comes back empty. I query them using the return Token.

Is this action not supported yet for a draft or maybe I'm doing something wrong?

**Comment by justlevine** [comment, marked as answer]
Hey @gmanu93 - is your File Upload field for single file uploads, or multiple?

[From GF's documentation:](https://docs.gravityforms.com/save-continue-gravity-forms/#h-how-save-and-continue-works)
> Single File Upload fields will not preserve any file added prior to the Save and Continue action. This is because single file fields are not validated as non-malicious until form submission is submitted. Thus when Save and Continue is used, any single file uploads attempted will be cleared and not preserved by the link. This behavior does not exist with Multi-File Upload fields, as they validate uploaded files at the time of upload, not form submission.

If you're using a Multi-file upload and still not getting the draft values back, please open a bug report, so we can get it fixed 😎

**Reply by gmanu93** [reply]
Thank you for replying so fast @justlevine ,

On the WP side, the field has Multi-file upload enable + Save and continue later enable. Sadly it still returning null.
I noticed that even if I use the form on the WordPress website, add images and press save and continue later, It doesn't store the files in the draft either or at least, it keeps coming back null on the query. I'll still be trying and discarding some options in the case is something I'm doing wrong, otherwise, I'll open a bug report. 

Thank you!

**Reply by justlevine** [reply]
If its happening in default WP, then its an issue with Gravity Forms, and not our plugin. I suggest you contact their support directly.

Good luck!
